src/Microsoft.DotNet.Cli.Utils/CommandResolver.cs

```diff
@@ -172,12 +172,14 @@ private static CommandSpec ConfigureCommandFromPackage(string commandName, strin
 
                 fileName = CoreHost.Path;
 
+                var depsArg = string.Empty;
+
                 if (depsPath != null)
                 {
-                    args = $"--tpaFile:\"{depsPath}\" {args}";
+                    depsArg = $"\"--depsFile:{depsPath}\" ";
                 }
 
-                args = $"\"{dllPath}\" {args}";
+                args = $"\"{dllPath}\" {depsArg}{args}";
             }
             else
             {
```

Comment:
Can you add a note to the spec (https://github.com/dotnet/cli/blob/master/Documentation/specs/corehost.md) that the managed application is not allowed to have a first argument `--depsFile` for the `coreconsole` mode? Or should we namespace this like `--corehost-depsfile`? I am fine with what you have.